backend/src/Designer/Controllers/OptionsController.cs

```diff
@@ -119,14 +120,15 @@ public async Task<ActionResult<List<Option>>> GetOptionsList(string org, string
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [Route("{optionsListId}")]
-    public async Task<ActionResult> CreateOrOverwriteOptionsList(string org, string repo, [FromRoute] string optionsListId, [FromBody] List<Option> payload, CancellationToken cancellationToken = default)
+    public async Task<ActionResult<Dictionary<string, List<Option>>>> CreateOrOverwriteOptionsList(string org, string repo, [FromRoute] string optionsListId, [FromBody] List<Option> payload, CancellationToken cancellationToken = default)
     {
         cancellationToken.ThrowIfCancellationRequested();
         string developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
 
         var newOptionsList = await _optionsService.CreateOrOverwriteOptionsList(org, repo, developer, optionsListId, payload, cancellationToken);
+        var optionListWithId = new Dictionary<string, List<Option>> { {optionsListId, newOptionsList} };
 
-        return Ok(newOptionsList);
+        return Ok(optionListWithId);
```

Comment:
Do we need this? Dont we know in FE what optionsListId we did the put request towards? 🤔

Reply:
FE?

Reply:
Yes and no, the onSuccess function does not know it. It's out of scope

Reply:
Can we pass it to the onSuccess? Think we are doing that elsewhere in the solution 🤔